python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -128,6 +135,25 @@ def notify_resource_usage(
     ) -> None:
         free_slots = self._actor_pool.num_free_slots()
         if input_queue_size > free_slots and under_resource_limits:
+            # The first time we request actors, we record the timestamp.
+            if self._actors_requested_ts is None:
+                self._actors_requested_ts = time.time()
+
+            # If the minimum requested actors are not available in a timely manner,
+            # raise a timeout.
+            exceeds_timeout = (
+                time.time() - self._actors_requested_ts
+                > DEFAULT_WAIT_FOR_MIN_ACTORS_SEC
+            )
+            if (
+                exceeds_timeout
+                and self._actor_pool.num_running_actors()
+                < self._autoscaling_policy.min_workers
```

Comment:
to be safe, can we also check `self._actor_pool.num_pending_actors() > 0`? So we are making sure we only throw exception when we are actually requesting actors but not able to get them.